Add vSphere endpoint options (#474)
kubev2v/forklift-documentation: documentation/modules/adding-source-provider.adoc

You can add a VMware vSphere source provider by using the {ocp} web console.

> In {project-short} versions 2.6 and later, you can migrate VMs from vCenter, from an ESX/ESXi server that is not managed by vCenter, or from an ESX/ESXi server that is managed by vCenter but does not go through vCenter.

Reviewer: @RichardHoch that's better but about the last part "but does not go through vCenter" - the idea is that MTV doesn't interact with vCenter in this situation (of a server that is managed by vCenter but the specified endpoint is that of the server). this phrasing makes it sound as the server doesn't go through vCenter

[IMPORTANT]
====
EMS enforcement is disabled for migrations with VMware vSphere source providers in order to enable migrations from versions of vSphere that are supported by {project-full} but do not comply with the 2023 FIPS requirements. Therefore, users should consider whether migrations from vSphere source providers risk their compliance with FIPS. Supported versions of vSphere are specified in xref:../master.adoc#compatibility-guidelines_mtv[Software compatibility guidelines].
====

.Prerequisites

* It is strongly recommended to create a VMware Virtual Disk Development Kit (VDDK) image in a secure registry that is accessible to all clusters. A VDDK image accelerates migration. For more information, see xref:../master.adoc#creating-vddk-image_mtv[Creating a VDDK image].
endif::[]
ifdef::rhv[]
= Adding {a-rhv} source provider

> You can add {a-rhv} source provider by using the {ocp } web console.

Reviewer: ```suggestion
You can add {a-rhv} source provider by using the {ocp} web console.
```

You can add a VMware vSphere source provider by using the {ocp} web console.

> In MTV versions 2.6 and later, you can also migrate VMs from standalone ESX/ESXi, or from ESX/ESXi servers that are managed by vCenter without going through vCenter.  This option is included in the procedure that follows.

Reviewer 2: ```suggestion
In {project-short} versions 2.6 and later, you can also migrate VMs from standalone ESX/ESXi, or from ESX/ESXi servers that are managed by vCenter without going through vCenter.  This option is included in the procedure that follows.
```

> In {project-short} versions 2.6 and later, you can also migrate VMs from standalone ESX/ESXi, or from ESX/ESXi servers that are managed by vCenter without going through vCenter.  This option is included in the procedure that follows.

Reviewer: we don't mention vCenter before so it may not be so obvious to understand what "you can also" refers to
I'm also not sure if for "standalone ESX/ESXi" is a common term
how about:
```suggestion
In {project-short} versions 2.6 and later, in addition to migrations from vCenter, you can also migrate VMs from standalone ESX/ESXi, i.e., an ESX/ESXi server that is not managed by vCenter, or from ESX/ESXi servers that are managed by vCenter without going through vCenter.  This option is included in the procedure that follows.
```

Author: If "standalone ESX/ESXi" doesn't sound good to you, then I'll avoid it.  Also the sentence beginning "This option is included ..." is not needed -- since I mentioned the option, I'd better include it in the documentation :) My bad.

. Specify the following fields:

* *Provider resource name*: Name of the source provider.

> * *Endpoint type*: Select the vSphere provider endpoint type. Options: *vCenter* or *ESXi*.

Reviewer: or maybe regarding the previous comment, this is the documentation of 2.6 so we can drop that statement above (or move it to the release notes) and add it here:
 ```suggestion
* *Endpoint type*: Select the vSphere provider endpoint type. Options: *vCenter* or *ESXi*. ESXi can be used for standalone servers, i.e.,  an ESX/ESXi server that is not managed by vCenter, or from ESX/ESXi servers that are managed by vCenter without going through vCenter.
```
what do you think?

Author: @ahadas , I made some changes based on your suggestions. Please review the PR again.

. Specify the following fields:

* *Provider resource name*: Name of the source provider.

> * *Endpoint type*: Select the vSphere provider endpoint type. Options: *vSphere* or *ESXi*.

Reviewer 3: it's vCenter or ESXi , @ahadas please keep me honest here

Author: Fixed.

* *URL*: URL of the SDK endpoint of the vCenter on which the source VM is mounted. Ensure that the URL includes the `sdk` path, usually `/sdk`. For example, `https://vCenter-host-example.com/sdk`. If a certificate for FQDN is specified, the value of this field needs to match the FQDN in the certificate.
* *VDDK init image*: `VDDKInitImage` path. It is strongly recommended to create a VDDK init image to accelerate migrations. For more information, see xref:../master.adoc#creating-vddk-image_mtv[Creating a VDDK image].

> * *Username*: vCenter user. For example, `user@vsphere.local`.

Reviewer 3: vCenter and ESXi may have different user name and password, so this should not make users think they need to use there vCenter credentials even if they connect to ESXi, on ESXi they should use the ESXi user+password and on vCenter they should use the vCenter user+password,  , @ahadas please keep me honest here
